### PR TITLE
Fix Google App Engine ImportError.

### LINKED ISSUE #594: Google App Engine ImportError: No module named msvcrt
Because Click was added to [Flask 0.11](http://flask.pocoo.org/docs/0.11/) for the [New Command Line Interface](http://flask.pocoo.org/docs/0.11/cli/), and Click imports the Windows only module [msvcrt](https://docs.python.org/2/library/msvcrt.html), and the [Google App Engine local development server](https://cloud.google.com/appengine/docs/python/tools/using-local-server) doesn't contain this module, an import error is thrown.

image

I've [created a workaround](https://github.com/gae-init/gae-init/pull/527) for the [gae-init project](https://github.com/gae-init/gae-init) (by modifying the App Engine config file), but it would be best to have a fix come directly from Flask/Click since there are likely many more people who will experience this issue when using Flask 0.11 with Google App Engine.

### PR DESCRIPTION
**Pull request by zacbir**
Fixes #594 

Recreated from PR #617

### CODE REVIEW

File: click/_compat.py [hunk 1]
```diff
@@ -7,8 +7,9 @@
 
 
 PY2 = sys.version_info[0] == 2
-WIN = sys.platform.startswith('win')
 CYGWIN = sys.platform.startswith('cygwin')
+APP_ENGINE = os.environ.get('SERVER_SOFTWARE','').startswith('Development')
```
Comment (ThiefMaster) [inline click/_compat.py]: space after `,`

Comment (ThiefMaster) [reply click/_compat.py]: Also, what's the full name of the App Engine `SERVER_SOFTWARE`? Starting with "Development" sounds extremely generic/vague.

Comment (zacbir) [reply click/_compat.py]: Can definitely fix the stylistic issue. We'll determine what the proper way of establishing a development version of GAE is. According to Google's own docs, they do something similar to determine whether we're running in a production environment (from: https://cloud.google.com/appengine/docs/standard/python/tools/using-local-server):

```
if os.getenv('SERVER_SOFTWARE', '').startswith('Google App Engine/'):
  # Production
else:
  # Local development server
```

**Comment by zacbir** [comment]
Sorry that review got clobbered. We're still looking for some GAE engineer guidance.

**Comment by zacbir** [comment]
Hey, @theacodes Any guidance on the best way to determine running under GAE locally?

According to Google's own docs, they do something similar to determine whether we're running in a production environment (from: https://cloud.google.com/appengine/docs/standard/python/tools/using-local-server):

```
if os.getenv('SERVER_SOFTWARE', '').startswith('Google App Engine/'):
  # Production
else:
  # Local development server
```

**Comment by theacodes** [comment]
See urllib3: https://github.com/urllib3/urllib3/blob/master/urllib3/contrib/appengine.py#L283-L305

**Comment by zacbir** [comment]
@ThiefMaster How's that look?

**Pull request merged**